browser/brave_content_browser_client_browsertest.cc

```diff
@@ -740,3 +745,92 @@ IN_PROC_BROWSER_TEST_F(BraveContentBrowserClientTest, CheckExpectedExtensions) {
   // Brave Extension background page should be disabled by default.
   EXPECT_FALSE(extensions::BackgroundInfo::HasBackgroundPage(brave_extension));
 }
+
+#if BUILDFLAG(IS_WIN)
+
+class WinBraveContentBrowserClientTest
+    : public BraveContentBrowserClientTest,
+      public ::testing::WithParamInterface<
+          base::test::ScopedOSInfoOverride::Type> {
+ public:
+  WinBraveContentBrowserClientTest() = default;
+  ~WinBraveContentBrowserClientTest() override = default;
+
+  bool GetShouldDoLearning(Browser* browser) {
+    bool value = false;
+    browser->tab_strip_model()
+        ->GetActiveWebContents()
+        ->GetShouldDoLearningForTesting(value);
+    return value;
+  }
+
+  bool IsWin11() const {
+    return GetParam() == base::test::ScopedOSInfoOverride::Type::kWin11Home;
+  }
+
+ private:
+  base::test::ScopedOSInfoOverride win_version_{GetParam()};
+};
+
+INSTANTIATE_TEST_SUITE_P(
+    ,
+    WinBraveContentBrowserClientTest,
+    ::testing::Values(base::test::ScopedOSInfoOverride::Type::kWin11Home,
+                      base::test::ScopedOSInfoOverride::Type::kWin10Home));
+
+IN_PROC_BROWSER_TEST_P(WinBraveContentBrowserClientTest, PRE_WindowsRecall) {
+  EXPECT_EQ(windows_recall::IsWindowsRecallAvailable(), IsWin11());
```

Comment:
browser tests are substantially slower and use far more resources than unit tests. Please do not use them for things that can be unit tested.

Reply:
Also this test could fail depending on the version of windows this test runs on if the logic in `IsWindowsRecallAvailable` is incorrect so it isn't an effective check

Reply:
@boocmp that is not correct. If you run on win11, you don't know if it is disabled on win10. This test only tells you that the behavior is correct on the version of windows you run it on

Reply:
To be clear I'm not saying that we shouldn't have any browser tests. We need a few to cover the full functionality including verifying that the scripts get injected and run correctly. However, there are many parts of this that can be effectively tested in unit tests for tab helper like choosing the correct files for a given domain, restore vs non-restore, disabled, etc...

Reply:
@boocmp In terms of the "restart" not being correct, the problem here is that the cache is tied to the lifetime of the profile which is not necessarily the same as the lifetime of the browser if you use more than one profile

Reply:
I can add a test with two profiles and check there is no "restart" requirement after you close and reopen this profile without restarting the browser if it really necessary

Reply:
now this does actually require browser restart so we can leave this